### PR TITLE
docs(testing): add test categorization rule and enforce import boundaries

### PR DESCRIPTION
**Pull request by hirotomoyamada**
Closes #

## AI used

- [ ] I did not use AI to create this PR.
- [x] (If there is no check above) I checked the generated content before submitting.

## Description

Adds a categorization rule that tells contributors which Vitest project a `packages/react` test belongs in (jsdom / all-browser / single-browser), and enforces the `#test` ↔ `#test/browser` import boundary at lint time so the file-name suffix introduced in #7013 cannot drift from the helper actually imported.

## Current behavior (updates)

- [.agents/rules/browser-testing.md](.agents/rules/browser-testing.md) lists the four file-name suffixes but does not say *when* to choose each, and the per-browser variants (`*.test.{chromium,firefox,webkit}.*`) have no usage guidance. Contributors have to infer the categorization from the existing inventory.
- The `#test` / `#test/browser` import boundary exists only as convention; nothing fails CI when a jsdom test imports `#test/browser` (or vice versa).
- `createNoRestrictedImportConfig` in [packages/workspace/src/oxlint/import.ts](packages/workspace/src/oxlint/import.ts) takes `[directory, ...patterns][]` and applies the patterns to `<directory>/**/*.{ts,cts,mts,tsx}`. It cannot target a subset of files inside a directory (e.g. only browser-test files).

## New behavior

- New [.agents/rules/test-categorization.md](.agents/rules/test-categorization.md): single source of truth for which file-name to choose, with a five-axis checklist (DOM read / event handling / observer / focus / browser API) and per-browser engine-branching guidance.
- [.agents/rules/browser-testing.md](.agents/rules/browser-testing.md) lead-in updated so each per-browser variant explains *when to use it*, and points at test-categorization.md for the full decision.
- [AGENTS.md](AGENTS.md) (= `CLAUDE.md` symlink) rule index gains a `Test Categorization` entry.
- `createNoRestrictedImportConfig` signature changes from `[directory, ...patterns][]` to `[files, ...patterns][]`, so callers pass the file glob list directly. This unlocks targeting test-file subsets.
- New `sharedUnitTestFiles` and `sharedBrowserTestFiles` exports in [packages/workspace/src/oxlint/shared.ts](packages/workspace/src/oxlint/shared.ts). `sharedTestFiles` is preserved as the union for future reuse.
- [packages/react/oxlint.config.ts](packages/react/oxlint.config.ts) wires two new overrides via the new signature: browser-test files are forbidden from importing `#test`, and unit-test (jsdom) files are forbidden from importing `#test/browser`.
- The signature change is propagated to all existing callers so the build stays green: [playgrounds/next/{app,pages}/oxlint.config.ts](playgrounds/next/) and [www/oxlint.config.ts](www/oxlint.config.ts).

## Is this a breaking change (Yes/No):

No. Public API unchanged. Only internal lint configuration and contributor-facing rules.

## Additional Information

Migration of the existing 141 `*.test.browser.{ts,tsx}` files in `packages/react/src/components/` is intentionally out of scope. That work happens in follow-up PRs grouped by component family, and uses this rule as the categorization reference.

### CODE REVIEW

**Review by claude-for-yamada-ui — APPROVED** [review]

**Review by codex-for-yamada-ui — APPROVED** [review]

File: .agents/rules/test-categorization.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## How to Choose
+
+- Choose browser mode (`*.test.browser.{ts,tsx}`) when any of the following applies:
+  - Logic that reads the DOM. For example, `getBoundingClientRect`, `getComputedStyle`, `scrollTop`, `scrollHeight`, `offsetHeight`, etc.
+  - Logic that handles events. For example, `onClick`, `onPointerDown`, etc.
+  - Logic that depends on observers. For example, `IntersectionObserver`, `ResizeObserver`, etc.
+  - Logic that depends on focus operations. For example, `focus`, `blur`, `activeElement`, etc.
+  - Logic that uses browser APIs. For example, `drag`, `clipboard`, `canvas`, `FileReader`, etc.
+- When the component or hook logic branches on `isWebKit`, `isSafari`, `isFirefox`, or `isChrome`, or the assertion differs by engine, choose `*.test.{webkit,firefox,chromium}.{ts,tsx}`.
+- Otherwise, when the assertion just checks element attributes (`displayName`, `className`, `role`, `aria-*`) and similar, choose jsdom (`*.test.{ts,tsx}`).
```
Comment (composer-for-yamada-ui) [inline .agents/rules/test-categorization.md:26]: **Issue:** Line 26 groups `displayName` with “element attributes” (`className`, `role`, `aria-*`). In typical Yamada UI tests `displayName` is asserted on the component export (`SomeComponent.displayName`), not on DOM elements returned from Testing Library queries, which mis-characterizes those tests as DOM-attribute-only checks versus component metadata checks.

**Fix:** Split wording so `displayName` is described explicitly as component/metadata assertions (matching patterns like component static properties), separately from DOM element attributes queried from rendered output (`role`, `aria-*`, `className`).

**Review by composer-for-yamada-ui — DISMISSED** [review]
- Accuracy: Line 26 groups `displayName` with “element attributes” (`className`, `role`, `aria-*`). In typical Yamada UI tests `displayName` is asserted on the component export (`SomeComponent.displayName`), not on DOM elements returned from Testing Library queries, which mis-characterizes those tests as DOM-attribute-only checks versus component metadata checks.

**Review by composer-for-yamada-ui — APPROVED** [review]
